agent/agent.go

```diff
@@ -1189,6 +1189,7 @@ func (a *Agent) logPanic(logger client.LoggerProducer, pErr, originalErr error,
 		logMsg := message.Fields{
 			"message":   "programmatic error: Evergreen agent hit panic",
 			"operation": op,
+			"host_id":   a.ec2InstanceID,
```

Comment:
While this is generally fine, it would be better to pass [the agent's host ID](https://github.com/evergreen-ci/evergreen/blob/70172791d8c8b43b56506502ea75aaef7d204f08/agent/agent.go#L59) here because 1. the EC2 instance ID only applies to EC2-spawnable hosts (so it won't be set for MacOS) and 2. we aren't guaranteed to be able to get `a.ec2InstanceID` since it's we have to `GET` it from an endpoint, whereas the host ID I linked is always populated when the agent starts since it's required to auth back to Evergreen.

Reply:
I kept the ec2 id (renamed to ec2_id), added the host_id and pod_id. I'm not sure if this has redundant id's (I haven't done much debugging with our agents) but looking at the options struct it seemed like it may help.

Lmk if they look good to you!

Reply:
The EC2 instance ID is the same as the host ID if it's an EC2 host, so I don't think we need `ec2_id` here. As for host ID vs pod ID, they're mutually exclusive, so I think maybe we can just add some conditional logic to only add them to the log if they're non-empty. Something like:
```go
logMsg := message.Fields { /* the original stuff... */ }
if a.opts.HostID != "" {
    logMsg["host_id"] = a.opts.hostID
}
if a.opts.PodID != "" {
    logMsg["pod_id"] = a.opts.PodID
}
```

Reply:
Ah makes sense! I thought they may be the same but wasn't sure. I'll add that logic in ty!